src/Opaleye/Internal/Sql.hs

```diff
@@ -108,6 +118,7 @@ baseTable ti columns = SelectFrom $
               , tables = [Table (HSql.SqlTable (PQ.tiSchemaName ti) (PQ.tiTableName ti))] }
 
 product :: NEL.NonEmpty Select -> [HPQ.PrimExpr] -> Select
+product (t NEL.:| []) [] = t
```

Comment:
Looks like a worthy change.  Filing this so we don't lose track of it: https://github.com/tomjaguarpaw/haskell-opaleye/issues/272